Dalamud/Interface/Internal/InterfaceManager.cs

```diff
@@ -248,14 +245,7 @@ internal set
         /// </summary>
         public void Dispose()
         {
-            // HACK: this is usually called on a separate thread from PresentDetour (likely on a dedicated render thread)
-            // and if we aren't already disabled, disposing of the scene and hook can frequently crash due to the hook
-            // being disposed of in this thread while it is actively in use in the render thread.
-            // This is a terrible way to prevent issues, but should basically always work to ensure that all outstanding
-            // calls to PresentDetour have finished (and Disable means no new ones will start), before we try to cleanup
-            // So... not great, but much better than constantly crashing on unload
-            this.Disable();
-            Thread.Sleep(500);
+            this.framework.RunOnFrameworkThread(this.Disable).Wait();
```

Comment:
setCursorHook is disabled in Disable(), yet we Dispose it immediately here... Dispose() automatically disables it, so aren't you double disposing?

Reply:
Double disable is acceptable via how Hook<...> is implemented, and it does Dispose them only once. Since we do not want game threads to have code running when we dispose the hook, we disable it from Framework.Tick, which is the time when the game shouldn't be running any of the hooked code. It is guaranteed that the game is the one to run our hook code for winapi hooks, as we're hooking by overwriting game's import table.

Reply:
Changed to make intentions clear.